### ISSUE TITLE
Fails to build successfully

### ISSUE THREAD

**Issue by jacek99** [madhead/docker-oracle-xe#6]
build script fails, unable to get it to run.
Running this on Fedora 22.

Database Configuration failed.  Look into /u01/app/oracle/product/11.2.0/xe/config/log for details
The command '/bin/sh -c /etc/init.d/oracle-xe configure responseFile=/u01/app/oracle/product/11.2.0/xe/config/scripts/xe.rsp' returned a non-zero code: 1

-------------------------------------------------------------------------------------------------------------

 ./build-wrapper.sh 
Sending build context to Docker daemon 317.3 MB
Step 0 : FROM centos:centos7
 ---> 0f73ae75014f
Step 1 : MAINTAINER Siarhei Krukau <siarhei.krukau@gmail.com>
 ---> Using cache
 ---> 03ad9f646087
Step 2 : RUN yum install -y libaio bc;     yum clean all
 ---> Using cache
 ---> 1d89a2ef7572
Step 3 : ADD rpm/oracle-xe-11.2.0-1.0.x86_64.rpm /tmp/
 ---> c962db0fc09e
Removing intermediate container bbabb38280ba
Step 4 : RUN yum localinstall -y /tmp/oracle-xe-11.2.0-1.0.x86_64.rpm;     rm -rf /tmp/oracle-xe-11.2.0-1.0.x86_64.rpm
 ---> Running in b2c3f917b77f
Loaded plugins: fastestmirror
Examining /tmp/oracle-xe-11.2.0-1.0.x86_64.rpm: oracle-xe-11.2.0-1.0.x86_64
Marking /tmp/oracle-xe-11.2.0-1.0.x86_64.rpm to be installed
Resolving Dependencies
--> Running transaction check
---> Package oracle-xe.x86_64 0:11.2.0-1.0 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

================================================================================
 Package       Arch       Version        Repository                        Size
================================================================================
Installing:
 oracle-xe     x86_64     11.2.0-1.0     /oracle-xe-11.2.0-1.0.x86_64     564 M

Transaction Summary
================================================================================
Install  1 Package

Total size: 564 M
Installed size: 564 M
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
/var/tmp/rpm-tmp.TXKdW1: line 257: [: 18446744073692774399: integer expression expected
/var/tmp/rpm-tmp.TXKdW1: line 271: [: 18446744073692774399: integer expression expected
  Installing : oracle-xe-11.2.0-1.0.x86_64                                  1/1 
Executing post-install steps...
You must run '/etc/init.d/oracle-xe configure' as the root user to configure the database.

  Verifying  : oracle-xe-11.2.0-1.0.x86_64                                  1/1 

Installed:
  oracle-xe.x86_64 0:11.2.0-1.0                                                 

Complete!
 ---> 01a8b11abeee
Removing intermediate container b2c3f917b77f
Step 5 : ADD config/xe.rsp config/init.ora config/initXETemp.ora /u01/app/oracle/product/11.2.0/xe/config/scripts/
 ---> 58de49f555ae
Removing intermediate container 45301e8533ce
Step 6 : RUN chown oracle:dba /u01/app/oracle/product/11.2.0/xe/config/scripts/*.ora                      /u01/app/oracle/product/11.2.0/xe/config/scripts/xe.rsp
 ---> Running in e36ef9deb039
 ---> 55d451b02d23
Removing intermediate container e36ef9deb039
Step 7 : RUN chmod 755        /u01/app/oracle/product/11.2.0/xe/config/scripts/*.ora                      /u01/app/oracle/product/11.2.0/xe/config/scripts/xe.rsp
 ---> Running in 6aca94a65f24
 ---> cfda6c449bec
Removing intermediate container 6aca94a65f24
Step 8 : ENV ORACLE_HOME /u01/app/oracle/product/11.2.0/xe
 ---> Running in 0754576fe36c
 ---> 76414163c17f
Removing intermediate container 0754576fe36c
Step 9 : ENV ORACLE_SID XE
 ---> Running in cc90798ae5a6
 ---> c5e4f0361308
Removing intermediate container cc90798ae5a6
Step 10 : ENV PATH $ORACLE_HOME/bin:$PATH
 ---> Running in d64deea495d5
 ---> 362ca4cd071f
Removing intermediate container d64deea495d5
Step 11 : RUN /etc/init.d/oracle-xe configure responseFile=/u01/app/oracle/product/11.2.0/xe/config/scripts/xe.rsp
 ---> Running in 8a0a60721110
/etc/init.d/oracle-xe: line 69: /etc/init.d/functions: No such file or directory

Oracle Database 11g Express Edition Configuration
-------------------------------------------------
This will configure on-boot properties of Oracle Database 11g Express 
Edition.  The following questions will determine whether the database should 
be starting upon system boot, the ports it will use, and the passwords that 
will be used for database accounts.  Press <Enter> to accept the defaults. 
Ctrl-C will abort.

Specify the HTTP port that will be used for Oracle Application Express [8080]:
/etc/init.d/oracle-xe: line 362: netstat: command not found
Specify a port that will be used for the database listener [1521]:
/etc/init.d/oracle-xe: line 405: netstat: command not found
Specify a password to be used for database accounts.  Note that the same
password will be used for SYS and SYSTEM.  Oracle recommends the use of 
different passwords for each database account.  This can be done after 
initial configuration:
Confirm the password:

Do you want Oracle Database 11g Express Edition to be started on boot (y/n) [y]:
Starting Oracle Net Listener...touch: cannot touch '/var/lock/subsys/listener': No such file or directory
Done
Configuring database...
Database Configuration failed.  Look into /u01/app/oracle/product/11.2.0/xe/config/log for details
The command '/bin/sh -c /etc/init.d/oracle-xe configure responseFile=/u01/app/oracle/product/11.2.0/xe/config/scripts/xe.rsp' returned a non-zero code: 1

**Comment by madhead** [comment]
So, what's in the log (`/u01/app/oracle/product/11.2.0/xe/config/log`) ?

**Comment by jacek99** [comment]
postScripts.log
-------------------------------------

CREATE OR REPLACE LIBRARY dbms_sumadv_lib AS '/u01/app/oracle/product/11.2.0/xe/lib/libqsmashr.so';
*
ERROR at line 1:
ORA-01034: ORACLE not available
Process ID: 0
Session ID: 0 Serial number: 0


BEGIN dbms_datapump_utl.replace_default_dir; END;

*
ERROR at line 1:
ORA-01034: ORACLE not available
Process ID: 0
Session ID: 0 Serial number: 0


commit
*
ERROR at line 1:
ORA-01034: ORACLE not available
Process ID: 0
Session ID: 0 Serial number: 0


create or replace directory XMLDIR as '/u01/app/oracle/product/11.2.0/xe/rdbms/xml'
*
ERROR at line 1:
ORA-01034: ORACLE not available
Process ID: 0
Session ID: 0 Serial number: 0


DROP DIRECTORY ORACLE_OCM_CONFIG_DIR
*
ERROR at line 1:
ORA-01034: ORACLE not available
Process ID: 0
Session ID: 0 Serial number: 0


DROP DIRECTORY ADMIN_DIR
*
ERROR at line 1:
ORA-01034: ORACLE not available
Process ID: 0
Session ID: 0 Serial number: 0


DROP DIRECTORY WORK_DIR
*
ERROR at line 1:
ORA-01034: ORACLE not available
Process ID: 0
Session ID: 0 Serial number: 0


BEGIN dbms_swrf_internal.cleanup_database(cleanup_local => FALSE); END;

*
ERROR at line 1:
ORA-01034: ORACLE not available
Process ID: 0
Session ID: 0 Serial number: 0

**Comment by jacek99** [comment]
postDBCreation.log
----------------------------------------

begin
*
ERROR at line 1:
ORA-01034: ORACLE not available
Process ID: 0
Session ID: 0 Serial number: 0



File created.

ORA-01034: ORACLE not available
ORA-27101: shared memory realm does not exist
Linux-x86_64 Error: 2: No such file or directory
ORA-03113: end-of-file on communication channel
ERROR:
ORA-03114: not connected to ORACLE


ERROR:
ORA-03114: not connected to ORACLE


ERROR:
ORA-03114: not connected to ORACLE

**Comment by jacek99** [comment]
So looks like Oracle cannot start up and we get ORACLE not available errors everywhere...

**Comment by wharsojo** [comment]
@jacek99 

I check on your logs, it seems you have some error and I already fix that error on my [fork](https://github.com/wharsojo/docker-oracle-xe/commit/2cbf38ac7f1f9946ce066d5b3152cd6713002cfb) 

I did create a [Pull Request](https://github.com/madhead/docker-oracle-xe/pull/7), need @madhead to review it.

Thanks,
Widi Harsojo.
[myboot2docker](https://github.com/wharsojo/myboot2docker)

**Comment by jacek99** [comment]
yes, some of those errors (like netstat) are fixed with your fork.

But I still get the same result at the end:

Do you want Oracle Database 11g Express Edition to be started on boot (y/n) [y]:
Starting Oracle Net Listener...Done
Configuring database...
Database Configuration failed.  Look into /u01/app/oracle/product/11.2.0/xe/config/log for details
The command '/bin/sh -c /etc/init.d/oracle-xe configure responseFile=/tmp/xe.rsp' returned a non-zero code: 1

**Comment by wharsojo** [comment]
I am using CentOS-7 and the build is fine:
![docker-oracle-xe](https://cloud.githubusercontent.com/assets/11621/10082058/7ce3f828-632a-11e5-9bb8-1793bfe55c5c.gif)
not sure what happen to your system.

**Comment by jacek99** [comment]
it's a vanilla Docker install on Fedora 22, so native Linux host. All other Dockerfiles I have work just fine.

Just Oracle XE is not.

**Comment by jacek99** [comment]
Found it.

The extra security protection of SELinux was stopping the XE configuration from completing.
Once I switched SELinux to permissive mode, it worked fine.

But one should not switch SELinux off, it is an important security protection, especially for a production box.

https://github.com/fedora-cloud/docker-selinux

I will re-enable it and try to tweak the Docker SELinux policy to figure out which fine grained setting is involved here and will let you know so it can be documented as a requirement if running this Dockerfile on Fedora or derivatives.
